packages/compass-import-export/src/modules/export.ts

```diff
@@ -362,16 +362,25 @@ export const runExport = ({
       dataService: { dataService },
     } = getState();
 
+    let fieldCount = 0;
+
     const query =
-      selectedFieldOption === 'select-fields'
+      exportFullCollection || aggregation
```

Comment:
Here's the fix mentioned in the description. We were actually using the query when exporting a full collection. Thanks to adding tests we catch these things :)